### PR TITLE
Indexing speedup

### PR DESCRIPTION
**Pull request by dojutsu-user**

### CODE REVIEW

File: readthedocs/search/management/commands/reindex_elasticsearch.py [hunk 1]
```diff
@@ -19,17 +18,27 @@ class Command(BaseCommand):
...
+            'document_class': document_class,
+            'index_name': index_name,
+        }
+
+        while not is_iterator_empty:
+            objects_id = []
+
+            try:
+                for _ in range(chunk_size):
+                    objects_id.append(qs_iterator.__next__().pk)
```
Comment (stsewd) [inline readthedocs/search/management/commands/reindex_elasticsearch.py]: You can just call `next(qs_iterator).pk`

Comment (davidfischer) [inline readthedocs/search/management/commands/reindex_elasticsearch.py]: This section could probably be simplified by using [`next()`](https://docs.python.org/3.6/library/functions.html#next) with a default value.

Comment (dojutsu-user) [reply readthedocs/search/management/commands/reindex_elasticsearch.py]: What should be the default?

File: readthedocs/search/management/commands/reindex_elasticsearch.py [hunk 2]
```diff
@@ -19,17 +18,27 @@ class Command(BaseCommand):
...
+        is_iterator_empty = False
+
+        data = {
+            'app_label': app_label,
+            'model_name': model_name,
+            'document_class': document_class,
+            'index_name': index_name,
+        }
+
+        while not is_iterator_empty:
```
Comment (stsewd) [inline readthedocs/search/management/commands/reindex_elasticsearch.py]: Actually you can just use `islice` from `itertools`

```python
from itertools import islice

objects_id = list(islice(qs_iterator, chunk_size))
```

File: readthedocs/search/management/commands/reindex_elasticsearch.py (hunk 1 above)
Comment (dojutsu-user) [reply readthedocs/search/management/commands/reindex_elasticsearch.py]: I think default is not required, current logic required to catch the exception.

File: readthedocs/search/management/commands/reindex_elasticsearch.py (hunk 2 above)
Comment (stsewd) [reply readthedocs/search/management/commands/reindex_elasticsearch.py]: And you can use https://docs.djangoproject.com/en/2.2/ref/models/querysets/#values-list to only get an iterator with only the ids

Comment (dojutsu-user) [reply readthedocs/search/management/commands/reindex_elasticsearch.py]: With this approach we will have a list of all the ids at once.
I think it is better to iterate over on generator object?

File: readthedocs/search/management/commands/reindex_elasticsearch.py [hunk 3]
```diff
@@ -19,17 +18,27 @@ class Command(BaseCommand):
...
+            objects_id = []
+
+            try:
+                for _ in range(chunk_size):
+                    objects_id.append(next(qs_iterator).pk)
```
Comment (ericholscher) [inline readthedocs/search/management/commands/reindex_elasticsearch.py]: I'd also like some logging here to be able to track progress. Something like `if pk % 5000 = 0: log.info('Total: pk')`

File: readthedocs/search/management/commands/reindex_elasticsearch.py (hunk 2 above)
Comment (stsewd) [reply readthedocs/search/management/commands/reindex_elasticsearch.py]: It returns a `QuerySet` of values, it's lazy https://stackoverflow.com/questions/37140426/does-django-queryset-values-list-return-a-list-object

File: readthedocs/search/management/commands/reindex_elasticsearch.py (hunk 3 above)
Comment (dojutsu-user) [reply readthedocs/search/management/commands/reindex_elasticsearch.py]: Done :+1:

File: readthedocs/search/management/commands/reindex_elasticsearch.py (hunk 2 above)
Comment (dojutsu-user) [reply readthedocs/search/management/commands/reindex_elasticsearch.py]: Ohhh...
But we also want to logging based on the pk.
Sorry... But I'm not able to think on how to use this method and to log also.
Do you have something on your mind?

Comment (stsewd) [reply readthedocs/search/management/commands/reindex_elasticsearch.py]: you can get the last pk from the list `last_pk = objects_id[-1] if objects_id else 0`

Comment (stsewd) [reply readthedocs/search/management/commands/reindex_elasticsearch.py]: Not so important anyway, current logic works too

Comment (ericholscher) [reply readthedocs/search/management/commands/reindex_elasticsearch.py]: I prefer the current implementation. It feels more explicit to me.

**Pull request merged**